### PR TITLE
chore(core): Update sdk-core and improve RBS generation for proto files

### PR DESCRIPTION
**Pull request by mjameswh**
## What changed

- Updated sdk-core to [d6454c](https://github.com/temporalio/sdk-core/commit/d6454c20800cf1ab53fca1831c9c39a277047515)
- Modify Rake proto:generate task to automatically fix errors in generated RBS files (they had been previously fixed by hand)
- Properly set `no_remote_activity` on core worker if no activity
- Pinned RBS dependency version to 2.8.x in order to resolve an incompatibility with steep 1.3.x, which was causing CI to hang.

### CODE REVIEW

File: sig/protos/temporal/api/enums/v1/namespace.rbs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module Temporalio
+module Temporal
```
Comment (cretz) [inline sig/protos/temporal/api/enums/v1/namespace.rbs:1]: Why did the namespace change on these? I think we want _everything_ in the SDK to be under `Temporalio`.

Comment (mjameswh) [reply sig/protos/temporal/api/enums/v1/namespace.rbs]: You were too early. 😄 I should have marked that PR as a draft.

File: sig/protos/google/protobuf/any.rbs [hunk 2]
```diff
@@ -0,0 +1,157 @@
+module Google
```
Comment (mjameswh) [inline sig/protos/google/protobuf/any.rbs:1]: I'm surprised this file is new. I might delete it.

File: sig/protos/temporal/api/batch/v1/message.rbs [hunk 3]
```diff
@@ -78,10 +82,14 @@ module Temporalio
         end
 
         # BatchOperationTermination sends terminate requests to batch workflows.
-        #  Keep the parameter in sync with temporal.api.workflowservice.v1.TerminateWorkflowExecutionRequest.
+        #  Keep the parameter in sync with temporalio.api.workflow_service.v1.TerminateWorkflowExecutionRequest.
```
Comment (mjameswh) [inline sig/protos/temporal/api/batch/v1/message.rbs:85]: This line should not have been modified

File: lib/temporalio/worker/sync_worker.rb [hunk 4]
```diff
@@ -21,66 +21,66 @@ def initialize(core_worker)
...
       end
 
       def record_activity_heartbeat(task_token, payloads)
-        proto = Coresdk::ActivityHeartbeat.new(
+        proto = Temporalio::Bridge::Api::CoreInterface::ActivityHeartbeat.new(
           task_token: task_token,
           details: payloads,
         )
-        encoded_proto = Coresdk::ActivityHeartbeat.encode(proto)
+        encoded_proto = Temporalio::Bridge::Api::CoreInterface::ActivityHeartbeat.encode(proto)
```
Comment (mjameswh) [inline lib/temporalio/worker/sync_worker.rb:58]: Verify why the extra `CoreInterface` here

**Review by cretz — APPROVED** [review]

File: lib/temporalio/worker/sync_worker.rb (hunk 4 above)
Comment (bergundy) [reply lib/temporalio/worker/sync_worker.rb:58]: I think if we use `Temporalio::Bridge::Core::` as the prefix for the core API it would be a bit nicer.

**Pull request merged**